functions/test/helloworld/log_test.rb

```diff
@@ -0,0 +1,46 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require "helper"
+
+describe "functions_log_helloworld" do
+  include FunctionsFramework::Testing
+
+  before do
+    @real_stdout = $stdout
+    @real_stderr = $stderr
+
+    $stdout = StringIO.new
+    $stderr = StringIO.new
+  end
+
+  after do
+    $stdout = @real_stdout
+    $stderr = @real_stderr
+  end
+
+  it "logs to stdout and stderr" do
+    load_temporary "helloworld/log.rb" do
+      request = make_get_request "http://example.com:8080"
+      response = call_http "log-helloworld", request
```

Comment:
You can use Minitest's `capture_subprocess_io` to capture stdout and stderr. (See `functions/test/helloworld/pubsub_test.rb` for an example.)

Reply:
That's much cleaner, thank you.